
 http://qa.mandriva.com/show_bug.cgi?id=28930

--- support-files/mysql-log-rotate.sh	2007-07-04 15:06:59.000000000 +0200
+++ support-files/mysql-log-rotate.sh.oden	2007-07-16 12:49:01.000000000 +0200
@@ -3,7 +3,7 @@
 # follows:
 #
 # [mysqld]
-# log-error=@localstatedir@/mysqld.log
+# log-error=/var/log/mysqld/mysqld.log
 #
 # In case the root user has a password, then you
 # have to create a /root/.my.cnf configuration file
@@ -18,7 +18,7 @@
 # ATTENTION: The /root/.my.cnf file should be readable
 # _ONLY_ by root !
 
-@localstatedir@/mysqld.log {
+/var/log/mysqld/mysqld.log {
         # create 600 mysql mysql
         notifempty
         daily
